ml_metadata/tools/mlmd_bench/proto/mlmd_bench.proto

```diff
@@ -158,6 +158,30 @@ message ReadTypesConfig {
   optional UniformDistribution maybe_num_ids = 2;
 }
 
+// Reads nodes: Artifacts / Executions / Contexts by their properties.
+message ReadNodesByPropertiesConfig {
+  enum Specification {
+    UNKNOWN = 0;
+    ARTIFACTS_BY_IDs = 1;
+    EXECUTIONS_BY_IDs = 2;
+    CONTEXTS_BY_IDs = 3;
+    ARTIFACTS_BY_TYPE = 4;
+    EXECUTIONS_BY_TYPE = 5;
+    CONTEXTS_BY_TYPE = 6;
+    ARTIFACT_BY_TYPE_AND_NAME = 7;
+    EXECUTION_BY_TYPE_AND_NAME = 8;
+    CONTEXT_BY_TYPE_AND_NAME = 9;
+    ARTIFACTS_BY_URIs = 10;
+  }
+  // Indicates which property (id, type, name, etc.)  
+  // should be used to get nodes (artifacts, executions, contexts).
+  optional Specification specification = 1;
+  // Specifies the number of ids or uris to be filled per request for 
+  // ARTIFACTS_BY_IDs / EXECUTIONS_BY_IDs / CONTEXTS_BY_IDs 
+  // or ARTIFACTS_BY_URIs specification. Modeled by a uniform distribution.
+  optional UniformDistribution maybe_num_queries = 2;
```

Comment:
num_of_parameters

+ docstring: When specification is ..., the num_of_parameters means ... 

+ Validation, check error if the specification should not set the num_of_params